### PR TITLE
Make `.secrets` creation mandatory

### PR DESCRIPTION
**Pull request by ESKYoung**
On reflection and discussion with @exfalsoquodlibet and @avisionh from comments in PR #10, we will always need to store some sort of credentials.

I've updated `govcookiecutter` with the following:

- Removed option to not create a `.secrets` file - this should now always happen
- Version-controlled here a `.secrets` file - **this will be ignored when the user creates a new template**
- Updated the documentation
- Added in caveats around the `detect-secrets` package, as highlighted through discussions with users
- Updated the package versions for the hooks to the latest release tag only

### CODE REVIEW

File: {{ cookiecutter.repo_name }}/CONTRIBUTING.md [hunk 1]
```diff
@@ -28,9 +28,9 @@ make requirements
 It is better to use the above make command, rather than `pip install -r requirements.txt` and `pre-commit install`, as
 the command will ensure your pre-commit hooks are up-to-date with any changes made.
 
-The pre-commit hooks are a security feature to ensure no secrets, large data files, and Jupyter notebooks are
-accidentally committed into the repository. For more information about the pre-commit hooks used in this repository,
-see the [documentation][docs-pre-commit-hooks].
+The pre-commit hooks are a security feature to ensure no secrets<sup>[1](#footnote-1)</sup>, large data files, and
+Jupyter notebooks are accidentally committed into the repository. For more information about the pre-commit hooks used
```
Comment (exfalsoquodlibet) [inline {{ cookiecutter.repo_name }}/CONTRIBUTING.md:32]:
```suggestion
Jupyter notebook outputs are accidentally committed into the repository. For more information about the pre-commit hooks used
```

Very minor, butI think this is a fairer descriptions of what the pre-commit hook does

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -28,9 +28,9 @@ make requirements
 It is better to use the above make command, rather than `pip install -r requirements.txt` and `pre-commit install`, as
 the command will ensure your pre-commit hooks are up-to-date with any changes made.
 
-The pre-commit hooks are a security feature to ensure no secrets, large data files, and output cells of Jupyter
-notebooks are accidentally committed into the repository. For more information about the pre-commit hooks used in this
-repository, see the [documentation][docs-pre-commit-hooks].
+The pre-commit hooks are a security feature to ensure no secrets<sup>[1](#footnote-1)</sup>, large data files, and
+Jupyter notebooks are accidentally committed into the repository. For more information about the pre-commit hooks used
```
Comment (exfalsoquodlibet) [inline CONTRIBUTING.md:32]:
```suggestion
Jupyter notebook outputs are accidentally committed into the repository. For more information about the pre-commit hooks used
```

Here again very minor point, but I think it is a fairer description of the purpose of this pre-commit hook.

**Review by exfalsoquodlibet — COMMENTED** [review]
Looks great, @ESKYoung !

A `.secrets` file is now created by default when using `make example` on the cloned repo branch. 

I could not test it via the interactive commands one gets when using 
```
cookiecutter https://github.com/ukgovdatascience/govcookiecutter
``` 
because this does not work with unmerged branches (e.g., `https://github.com/ukgovdatascience/govcookiecutter/tree/make_secrets_mandatory`) unless you know something I do not know - happy to test if so 😄

**Comment by ESKYoung** [comment]
> I could not test it via the interactive commands one gets when using
> 
> ```
> cookiecutter https://github.com/ukgovdatascience/govcookiecutter
> ```
> 
> because this does not work with unmerged branches (e.g., `https://github.com/ukgovdatascience/govcookiecutter/tree/make_secrets_mandatory`) unless you know something I do not know - happy to test if so 😄

Thanks for the review @exfalsoquodlibet - I'll respond to the rest of the comments tomorrow, but to do this, `git clone` the repository locally (if you haven't already), and `git checkout` the appropriate branch for review/testing. Then open your terminal, and `cd` to the folder **above** your cloned repository, and run:

```bash
cookiecutter govcookiecutter/
```

This should run `cookiecutter` on the locally checked-out code!

**Comment by exfalsoquodlibet** [comment]
Tested by running `cookiecutter` on the locally checked-out code! Works as expected 🎉 

Thanks a lot for explaining that. I had cloned the repo and checkout the branch etc. of course but I though it was not possible to run `cookiecutter` on local code - which does not make sense now that I think about it! 😄

**Review by exfalsoquodlibet — APPROVED** [review]

**Comment by ESKYoung** [comment]
Great thanks for checking @exfalsoquodlibet! RE: running on different branches, it turns out there's also a `--checkout` option for `cookiecutter` - see the [docs](https://cookiecutter.readthedocs.io/en/1.7.2/usage.html#works-directly-with-git-and-hg-mercurial-repos-too).

**Pull request merged**